src/encoded/auth0.py

```diff
@@ -96,6 +93,40 @@ def forget(self, request):
         return []
 
 
+def signup(request):
+    """
+    Create new user.
+
+    :param request: Pyramid request object
+    """
+    domain = 'encode.auth0.com'
+    url = "https://{domain}/userinfo?access_token={access_token}".format(domain=domain, access_token=request.json['accessToken'])
+    user_data_request = requests.get(url)
+
+    if user_data_request.status_code != 200:
+        raise HTTPBadRequest(explanation='Could not get user data')
+
+    users = request.registry[COLLECTIONS]['user']
+    user_data = user_data_request.json()
+
+    if not user_data['email_verified']:
+        raise HTTPBadRequest(explanation='Unverified e-mail')
+
+    user_info = {
+        'email': user_data['email'],
+        'first_name': user_data.get('given_name', '') or user_data.get('first_name', '') or user_data.get('name', ''),
```

Comment:
So someone could make an account without a name?

Reply:
Great point. Let me think about this too. Github just provides a name object; no first or last name. The ENCODE db does not blow up if there is no name. So, I tried to fine a good solution given this. Since you are raising a flag, my solution is not clear and may need to be revised.